numpy/core/fromnumeric.py

```diff
@@ -70,11 +70,29 @@ def take(a, indices, axis=None, out=None, mode='raise'):
     using arrays); however, it can be easier to use if you need elements
     along a given axis.
 
+    When axis is not None this is equivalent to the following use of `ndindex`,
+    which sets each of ``ii``, ``jj``, and ``kk`` to a tuple of indices::
+
+        Ni, Nk = a.shape[:axis], a.shape[axis+1:]
+        Nj = indices.shape
+        for ii in ndindex(Ni):
+            for jj in ndindex(Nj):
+                for kk in ndindex(Nk):
+                    out[ii + jj + kk] = a[ii + (indices[jj],) + kk]
```

Comment:
Alternatively:

```
out[(*ii, *jj, *kk)] = a[(*ii, indices[jj], *kk)]
```

It's a shame the unpacking generalizations don't apply to tuples within indexing brackets

Reply:
Also, while we are using `s_`, another way to express take is 

```
arr[(s_[:],)*axis + (indices, Ellipsis)]
```

which is essentially what my comment in the intro of the docstring was saying about `a[:,:,:,indices,...]`. 

What about adding something like this to the docstring? :
```
     This function does the same thing as "fancy" indexing (indexing arrays
     using arrays); however, it can be easier to use if you need elements
     along a given axis and supports `out` and `mode` arguments.

     That is, a call such as `np.take(arr, indices, axis=3)` is equivalent to 
     `arr[:,:,:,indices,...]`. More generally `take` is equivalent to the following use of `ndindex`
     which sets each of ``ii``, ``jj``, and ``kk`` to a tuple of indices::
```

Reply:
Added, with some rephrasing